treewide: Fix links in the docs and cleanup many boards (#22257)
RIOT-OS/RIOT: boards/stm32f3discovery/doc.md

Reviewer: ```suggestion
```

## Overview

> The STM32F3-discovery is cheap evaluation board designed by ST for pushing

Reviewer: ```suggestion
The STM32F3Discovery is cheap evaluation board designed by ST for pushing
```

| Magnetometer   | LSM303DLHC    | no        | planned |
| Gyroscope  | L3GD20/I3G4250D | yes        | |

## Flashing the device

The STM32F3discovery board includes an on-board ST-LINK V2 programmer. The
easiest way to program the board is to use OpenOCD. Once you have installed
OpenOCD (look [here](https://doc.riot-os.org/misc/openocd/) for
installation instructions), you can flash the board simply by typing

```
make flash
```

and debug via GDB by simply typing

```
make debug
```

> 

Reviewer: ```suggestion
## Flashing the Board

A detailed description about the flashing process can be found on the
[guides page](https://guide.riot-os.org/board_specific/stm32/).
The board name for the STM32F3discovery is `stm32f3discovery`.

```

## Supported Toolchains

For using the STM32F3discovery board we strongly recommend the usage of the
[GNU Tools for ARM Embedded Processors](https://launchpad.net/gcc-arm-embedded)
toolchain.

> 

Reviewer: ```suggestion
```

We usually get rid of this section, as it is outdated and misleading.
